### ISSUE TITLE
Agent deployed with Operator and default configuration does not work out of the box

### ISSUE THREAD

**Issue by sjdrc** [grafana/agent#1601]
After installing the operator from helm chart and following https://grafana.com/docs/agent/latest/operator/custom-resource-quickstart/#step-1-deploy-grafanaagent-resource, the `grafana-agent-0` pod fails to initialise with the following message `flag provided but not defined: -reload-port`. I see that this option has been removed for v0.24, and the operator does not support versions below v0.24, so I'm not really sure why the operator is configuring the StatefulSet with that flag.

**Comment by sjdrc** [comment]
After deleting everything and re-deploying but specifying v0.24 in the values for grafana-agent-operator, I am now running into the following:
```
2022-04-13 06:54:15.215613 I | error loading config file /var/lib/grafana-agent/config/agent.yml: yaml: unmarshal errors:
  line 7: field username_file not found in type config.plain
```

**Comment by flokli** [comment]
The helm chart has been updated in https://github.com/grafana/helm-charts/pull/1235, but yeah, with v0.24.0 I still get these issues.

**Comment by sjdrc** [comment]
Thanks for the prompt reply and fix!

> While we fix the issues, you may wish to deploy v0.23.0 of the operator and agent. We'll release 0.24.1 with fixes this week.
So specifically what version of the operator chart should I install? I was using 0.1.7 of the chart and v0.23.0 of the agent when I got the error in the OP.
